pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -122,21 +126,110 @@ func haveOverlap(a1, a2 []string) bool {
+func (pl *VolumeRestrictions) isReadWriteOncePodAccessModeConflict(cycleState *framework.CycleState, pod *v1.Pod) *framework.Status {
 	nodeInfos, err := pl.nodeInfoLister.NodeInfos().List()
 	if err != nil {
 		return framework.NewStatus(framework.Error, "error while getting node info")
 	}
+	pvcKeys, status := pl.keysForPVCWithRWOP(pod)
+	if status != nil {
+		return status
+	}
+	var nodesMatchPVCWithRWOP []*v1.Node
+	ctx, cancel := context.WithCancel(context.Background())
+	processNode := func(i int) {
+		nodeInfo := nodeInfos[i]
+		pvcKeysNumber := uint32(len(pvcKeys))
+		for key := range pvcKeys {
+			refCount := nodeInfo.PVCRefCounts[key]
+			if refCount != 0 {
+				atomic.AddUint32(&pvcKeysNumber, ^uint32(0))
+				if pvcKeysNumber == 0 {
+					nodesMatchPVCWithRWOP = append(nodesMatchPVCWithRWOP, nodeInfo.Node())
+					cancel()
+				}
+			}
+		}
+	}
+	pl.parallelizer.Until(ctx, len(nodeInfos), processNode)
+	// update cyclestate
+	p := &preFilterState{
+		nodesMatchPVC: nodesMatchPVCWithRWOP,
+		keysWithRWOP:  pvcKeys,
+	}
+	cycleState.Write(preFilterStateKey, p)
+	return nil
+}
 
+func (pl *VolumeRestrictions) PreFilterExtensions() framework.PreFilterExtensions {
+	return pl
+}
+
+func (pl *VolumeRestrictions) AddPod(ctx context.Context, cycleState *framework.CycleState, podToSchedule *v1.Pod, podInfoToAdd *framework.PodInfo, nodeInfo *framework.NodeInfo) *framework.Status {
+	state, err := getPreFilterState(cycleState)
+	if err != nil {
+		return framework.AsStatus(err)
+	}
+
+	pvcKeys, status := pl.keysForPVCWithRWOP(podToSchedule)
+	if status.IsUnschedulable() {
+		return status
+	}
+	pvcKeysNumber := len(pvcKeys)
+	for key := range pvcKeys {
+		refCount := nodeInfo.PVCRefCounts[key]
+		if refCount != 0 {
+			pvcKeysNumber--
+		}
+	}
+	if pvcKeysNumber == 0 {
+		state.nodesMatchPVC = append(state.nodesMatchPVC, nodeInfo.Node())
+	}
+	state.keysWithRWOP = pvcKeys.Union(state.keysWithRWOP)
+	return nil
+}
+
+func (pl *VolumeRestrictions) RemovePod(ctx context.Context, cycleState *framework.CycleState, podToSchedule *v1.Pod, podInfoToRemove *framework.PodInfo, nodeInfo *framework.NodeInfo) *framework.Status {
+	state, err := getPreFilterState(cycleState)
+	if err != nil {
+		return framework.AsStatus(err)
+	}
+	pvcKeys, status := pl.keysForPVCWithRWOP(podToSchedule)
+	if status.IsUnschedulable() {
+		return status
+	}
+	flag := false
+	for key := range pvcKeys {
+		refCount := nodeInfo.PVCRefCounts[key]
+		if refCount != 0 {
+			flag = true
+			break
+		}
+	}
+	if flag {
+		for i := range state.nodesMatchPVC {
+			if state.nodesMatchPVC[i].Name == nodeInfo.Node().Name {
+				state.nodesMatchPVC = append(state.nodesMatchPVC[:i], state.nodesMatchPVC[i+1:]...)
+			}
+		}
+	}
+	state.keysWithRWOP = state.keysWithRWOP.Difference(pvcKeys)
+	return nil
+}
+
+// Collect a pod in the form of namespace+pvcName with pvc whose accessmode is ReadWriteOncePod.
```

Comment:
It looks like this method is not collecting pods, but collecting the names of the PVCs on the pod with ReadWriteOncePod access mode

Reply:
Sorry, I was careless